public/app/percona/inventory/Tabs/Nodes.tsx

```diff
@@ -79,81 +83,95 @@ export const NodesTab = () => {
-                  {selected.length === 1 ? 'node' : 'nodes'}?
-                </h4>
-                <FormElement
-                  dataTestId="form-field-force"
-                  label="Force mode"
-                  element={
-                    <CheckboxField
-                      name="force"
-                      label={'Force mode is going to delete all ' + 'agents and services associated with the nodes'}
-                    />
-                  }
+    <Page navModel={navModel}>
+      <Page.Contents>
+        <PermissionLoader
+          featureSelector={() => true}
+          renderError={() => null}
+          renderSuccess={() => (
+            <div className={styles.tableWrapper}>
+              <div className={styles.actionPanel}>
+                <Button
+                  size="md"
+                  disabled={selected.length === 0}
+                  onClick={() => {
+                    setModalVisible(!modalVisible);
+                  }}
+                  icon="trash-alt"
+                  variant="destructive"
+                  className={styles.destructiveButton}
+                >
+                  Delete
+                </Button>
+              </div>
+              <Modal
+                title={
+                  <div className="modal-header-title">
+                    <span className="p-l-1">Confirm action</span>
+                  </div>
+                }
+                isOpen={modalVisible}
+                onDismiss={() => setModalVisible(false)}
+              >
+                <Form
+                  onSubmit={() => {}}
+                  render={({ form, handleSubmit }) => (
+                    <form onSubmit={handleSubmit}>
+                      <>
+                        <h4 className={styles.confirmationText}>
+                          Are you sure that you want to permanently delete {selected.length}{' '}
+                          {selected.length === 1 ? 'node' : 'nodes'}?
+                        </h4>
+                        <FormElement
+                          dataTestId="form-field-force"
+                          label="Force mode"
+                          element={
+                            <CheckboxField
+                              name="force"
+                              label={
+                                'Force mode is going to delete all ' + 'agents and services associated with the nodes'
+                              }
+                            />
+                          }
+                        />
+                        <HorizontalGroup justify="space-between" spacing="md">
+                          <Button variant="secondary" size="md" onClick={() => setModalVisible(false)}>
+                            Cancel
+                          </Button>
+                          <Button
+                            size="md"
+                            onClick={() => {
```

Comment:
Wouldn't it be better to have this as a callback?

Reply:
Indeed 👍